pandas/tests/indexes/test_base.py

```diff
@@ -663,6 +663,15 @@ def test_intersection(self):
         intersect = first.intersection(second)
         assert intersect.name is None
 
+    def test_intersect_str_dates(self):
```

Comment:
in reality, there are a bunch of specific tests here *just* on ``Index`` which are one-offs and not generally testing things. but ok.